bundles/org.openhab.binding.http/src/main/java/org/openhab/binding/http/internal/http/RefreshingUrlCache.java

```diff
@@ -76,9 +75,25 @@ public RefreshingUrlCache(ScheduledExecutorService executor, RateLimitedHttpClie
         this.httpContentType = httpContentType;
         this.httpStatusListener = httpStatusListener;
         fallbackEncoding = thingConfig.encoding;
+    }
 
-        future = executor.scheduleWithFixedDelay(this::refresh, 1, thingConfig.refresh, TimeUnit.SECONDS);
-        logger.trace("Started refresh task for URL '{}' with interval {}s", url, thingConfig.refresh);
+    public void start(ScheduledExecutorService executor, int refreshTime) {
+        if (future != null) {
+            logger.warn("Starting refresh task requested but it is already started. This is bug.");
+            return;
+        }
+        future = executor.scheduleWithFixedDelay(this::refresh, 1, refreshTime, TimeUnit.SECONDS);
+        logger.trace("Started refresh task for URL '{}' with interval {}s", url, refreshTime);
+    }
+
+    public void stop() {
+        // clearing all listeners to prevent further updates
+        consumers.clear();
+        ScheduledFuture<?> future = this.future;
+        if (future != null) {
+            future.cancel(false);
```

Comment:
I'm not sure how this should be handled, but the future is being cancelled without being interrupted, and a reference is kept since `this.future` is not set to null.  Since it's called by `HttpThingHandler.dispose`, I suppose we should actually both interrupt it and clear the reference?
```suggestion
            future.cancel(true);
            this.future = null;
```

Otherwise if started again you would run into the "this is a bug" warning logged in `start`, and it won't be rescheduled.

Reply:
No, a new `RefreshingUrlCache` is created when the thing is re-initialized, so this is not an issue.

Reply:
Okay, was not sure about the lifecycle, so nullifying becomes unimportant then. Still, shouldn't it be interrupted on `dispose`, so we don't have a job still running?

Reply:
I don't think it makes much difference, because the requests are cancelled anyway, but done.